src/nifs/vanilla/mod.rs

```diff
@@ -379,5 +380,32 @@ impl<C: CurveAffine> VerifyAccumulation<C> for VanillaFS<C> {
     }
 }
 
+/// As part of the vanilla folding scheme, we use the values in the zero instance of the column for
+/// consistency between folding steps
+///
+/// - X0 is
+///     initializing value
+///     or
+///     hash of the state at the end of previous folding step
+/// - X1 is a hash of the state at the end of the current folding step
+pub trait GetConsistencyMarkers<F> {
+    // TODO #329 Remove Option
+    fn get_consistency_markers(&self) -> Option<[F; 2]>;
+}
+
+impl<C: CurveAffine> GetConsistencyMarkers<C::ScalarExt> for PlonkInstance<C> {
+    fn get_consistency_markers(&self) -> Option<[C::ScalarExt; 2]> {
+        // TODO #329 Remove clone
+        self.instance.clone().try_into().ok()
+    }
+}
+
+impl<C: CurveAffine> GetConsistencyMarkers<C::ScalarExt> for RelaxedPlonkInstance<C> {
+    fn get_consistency_markers(&self) -> Option<[C::ScalarExt; 2]> {
+        // TODO #329 Remove clone
+        self.instance.clone().try_into().ok()
```

Comment:
since we are going to have multiple instances, one possible way is to put this [X0, X1] to be the first instance. i.e. `self.instances[0].clone()`
And ensure this when we create configuration of the circuit